tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -900,172 +833,24 @@ export class WebGPUBackend extends KernelBackend {
     return output;
   }
 
-  private getFromPixelTextureLayout(useImport: boolean): WebGPULayout {
-    if (useImport) {
-      if (this.fromPixelImportTextureLayout === null) {
-        this.fromPixelImportTextureLayout =
-            this.createFromPixelTextureLayout(true /* useImport */);
-      }
-      return this.fromPixelImportTextureLayout;
-    }
-
-    if (this.fromPixelTextureLayout === null) {
-      this.fromPixelTextureLayout =
-          this.createFromPixelTextureLayout(false /* useImport */);
-    }
-    return this.fromPixelTextureLayout;
-  }
-
-  private createFromPixelTextureLayout(useImport: boolean): WebGPULayout {
-    const bindGroupLayoutEntries: GPUBindGroupLayoutEntry[] = [];
-    // Output buffer binding layout.
-    bindGroupLayoutEntries.push({
-      binding: 0,
-      visibility: GPUShaderStage.COMPUTE,
-      buffer: {type: 'storage' as const }
-    });
-    // Input texture binding layout.
-    if (useImport) {
-      bindGroupLayoutEntries.push({
-        binding: 1,
-        visibility: GPUShaderStage.COMPUTE,
-        externalTexture: {},
-      });
-    } else {
-      bindGroupLayoutEntries.push(
-          {binding: 1, visibility: GPUShaderStage.COMPUTE, texture: {}});
-    }
-    // Uniform buffer binding layout.
-    bindGroupLayoutEntries.push(
-        {binding: 2, visibility: GPUShaderStage.COMPUTE, buffer: {}});
-    const fromPixelBindGroupLayout =
-        this.device.createBindGroupLayout({entries: bindGroupLayoutEntries});
-    const fromPixelPipelineLayout = this.device.createPipelineLayout(
-        {bindGroupLayouts: [fromPixelBindGroupLayout]});
-    return {
-      bindGroupLayout: fromPixelBindGroupLayout,
-      pipelineLayout: fromPixelPipelineLayout
-    };
-  }
-
-  private copyExternalImageToTexture(
-      externalImage: ExternalImage, outShape: number[]): GPUTextureView {
-    const textureUsage = GPUTextureUsage.COPY_DST |
-        GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.TEXTURE_BINDING;
-    const textureFormat = 'rgba8unorm' as GPUTextureFormat;
+  copyExternalImageToTexture(externalImage: ExternalImage, outShape: number[]):
+      TextureInfo {
+    const usage = GPUTextureUsage.COPY_DST | GPUTextureUsage.RENDER_ATTACHMENT |
+        GPUTextureUsage.TEXTURE_BINDING;
+    const format = 'rgba8unorm' as GPUTextureFormat;
     const texture = this.textureManager.acquireTexture(
-        outShape[1], outShape[0], textureFormat, textureUsage);
-    const externalResource = texture.createView();
+        outShape[1], outShape[0], format, usage);
 
     this.queue.copyExternalImageToTexture(
         {source: externalImage}, {texture}, [outShape[1], outShape[0]]);
 
-    const textureInfo = {
-      width: outShape[1],
-      height: outShape[0],
-      format: textureFormat,
-      usage: textureUsage,
+    return {
+      width: externalImage.width,
```

Comment:
Be careful that `externalImage.width` may be not equal to `outShape[1]`

Reply:
done